web/src/tableUtils.ts

```diff
@@ -0,0 +1,12 @@
+export function formatNumber(
+  number: number,
+  precision?: number,
+  threshold?: number
+): string {
+  threshold = threshold ? threshold : 4
+  if (number < 10 ** -threshold || number > 10 ** threshold) {
+    return number.toExponential(precision ? precision : 4).replace('e', 'E')
+  } else {
+    return number.toPrecision(precision ? precision : 4).toString()
```

Comment:
In JS, you can take default values like this
`function myFunc(myBool: boolean | undefined = true)`
Also, instead of `precision ? precision : 4` you can do `precision ?? 4`, though if you use default values this it not necessary anymore